src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,51 @@
 package core.basesyntax;
 
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        DateFormat dateFormat = new SimpleDateFormat("dd.MM.yyyy");
+        Date fromDate;
+        Date toDate;
+
+        try {
+            fromDate = dateFormat.parse(dateFrom);
+            toDate = dateFormat.parse(dateTo);
+        } catch (ParseException e) {
+            return "Invalid date format!";
+        }
+
+        StringBuilder report = new StringBuilder();
+        report.append("Report for period ").append(dateFrom)
+                .append(" - ").append(dateTo).append("\n");
```

Comment:
newline symbols are OS dependent, to deal with this use System.lineSeparator() it's the only correct way to go in this situation